src/BlockTypes/AbstractBlock.php

```diff
@@ -41,4 +48,33 @@ public function register_block_type() {
 			)
 		);
 	}
+
+	/**
+	 * Will enqueue any editor assets a block needs to load
+	 */
+	public function enqueue_editor_assets() {
+		$this->enqueue_data_once();
+	}
+
+
+	/**
+	 * Ensures the enqueue_data method is only called once in a request.
+	 */
+	public function enqueue_data_once() {
+		static $has_run = false;
```

Comment:
I'm not strong in PHP, but as far as I understand, static member variable is linked to the class, not instances of that class, so this will be linked to `AbstractBlock` not classes that it inherits? if that’s the case, then it will only work for the first block that runs it, but I don’t think this is the case because I tried adding both cart and checkout and nothing happend

Reply:
Ya actually this logic I added is flawed in numerous ways. I'm going to remove it and try something else.

Reply:
I've removed it and created #2077 to handle in a followup.  The way this is implemented currently doesn't really do anything.